sklearn/metrics/tests/test_pairwise.py

```diff
@@ -56,6 +57,16 @@ def test_pairwise_distances():
     S = pairwise_distances(X, Y, metric="euclidean")
     S2 = euclidean_distances(X, Y)
     assert_array_almost_equal(S, S2)
+    # Euclidean dist. (masked) should be equivalent to calling the function.
+    X = rng.random_sample((5, 4))
+    S = pairwise_distances(X, metric="masked_euclidean")
+    S2 = masked_euclidean_distances(X)
+    assert_array_almost_equal(S, S2)
+    # Euclidean distance, with Y != X.
```

Comment:
I would only keep this small test, and its point would be to check that `pairwise_distances` did not perform any unnecessary finiteness validation.